### ISSUE TITLE
`+` node should use `convolve` from Distributions.jl

### ISSUE THREAD

**Issue by bvdmitri** [biaslab/ReactiveMP.jl#72]
```
help?> convolve
search: convolve ConcurrencyViolationError

  convolve(d1::Distribution, d2::Distribution)

  Convolve two distributions and return the distribution corresponding to the sum of
  independent random variables drawn from the underlying distributions.

  Currently, the function is only defined in cases where the convolution has a closed
  form. More precisely, the function is defined if the distributions of d1 and d2 are
  the same and one of

    •  Bernoulli

    •  Binomial

    •  NegativeBinomial

    •  Geometric

    •  Poisson

    •  Normal

    •  Cauchy

    •  Chisq

    •  Exponential

    •  Gamma

    •  MvNormal

  External links: List of convolutions of probability distributions on Wikipedia
  (https://en.wikipedia.org/wiki/List_of_convolutions_of_probability_distributions)
```

**Comment by bvdmitri** [comment]
This probably would also need a specific `[compat]` version for Distributions.jl

**Comment by bartvanerp** [comment]
Implemented by a9841d4 and a5f72c8 in #69.

Currently the normal distribution resorts to specialized methods, as the `convolve` function accesses properties of the distribution, e.g. `dist.Σ`, whereas this property might not always be available. One way to solve this is to extend the univariate and multivariate normal distributions with properties, similarly as in https://github.com/JuliaLang/julia/blob/3bf9d1773144bc4943232dc2ffaac307a700853d/stdlib/LinearAlgebra/src/cholesky.jl#L497-L509. This might actually be a nice property to have in order to make the package more user-friendly. Just let me know what you think of it.

Not sure about the previous `[compat]` comment.

**Comment by bvdmitri** [comment]
Personally I don't like overriding `getproperty` function as for me it seems too dangerous and "dynamic".

> This might actually be a nice property to have in order to make the package more user-friendly.

As I understand this considered to be non-Julian approach (though even Julia Base uses it, but I assume this is just a very old code you shared). We already support `cov()` function for every Gaussian distribution, so there is no need for extra `dist.Σ` imo. And this is exactly the problem here, you cannot easily "add" properties to your structure, but you can easily add new methods to existing functions like `cov(::MyCustomPrettyGaussian)`. 

We probably should just define `Distributions.convolve` method for our own distributions, smth like:

```julia

import Distributions: convolve

function Distributions.convolve(left::UnivariateNormalDistributionsFamilty, right::UnivariateNormalDistributionsFamily)
    cleft  = convert(NormalMeanVariance, left)
    cright = convert(NormalMeanVariance, right)
    return NormalMeanVariance(mean(cleft) + mean(cright), var(cleft) + var(cright))
end
```

And in the same way for `MultivariateNormalDistributionsFamily`. We can put it in `distributions/normal.jl`. This file already has some "common" methods defined for the whole family.

> Not sure about the previous [compat] comment.

We have a `[compat]` section in our Project.toml with minimal versions of packages required. AFAIK `convolve` has been added recently (or at least documented recently) - we should double-check that `convolve` function exists in the `[compat]`-ible version of Distributions.jl. We can do it later when we will merge your PR. Just something to keep in mind.

**Comment by bvdmitri** [comment]
@bartvanerp  or instead of `convert` in my code snippet we could just use `mean_var()`. I think `mean_var()` is a better approach.

**Comment by bvdmitri** [comment]
Another benefit of our custom `convolve` function would be that we can use it in other places in our code. However I don't recall we ever needed it in any other place....

**Comment by bvdmitri** [comment]
Fixed in #69
